ctpolicy/applepolicy_test.go

```diff
@@ -88,3 +117,47 @@ func TestCheckApplePolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckApplePolicy2(t *testing.T) {
+	tests := []struct {
+		name string
+		cert *x509.Certificate
+		want LogPolicyData
+	}{
+		{
+			name: "Short",
+			cert: getTestCertPEMShort(),
+			want: wantedAppleGroups2(2),
+		},
+		{
+			name: "2-year",
+			cert: getTestCertPEM2Years(),
+			want: wantedAppleGroups2(3),
+		},
+		{
+			name: "3-year",
+			cert: getTestCertPEM3Years(),
+			want: wantedAppleGroups2(4),
+		},
+		{
+			name: "Long",
+			cert: getTestCertPEMLongOriginal(),
+			want: wantedAppleGroups2(5),
+		},
+	}
+
+	var policy AppleCTPolicy
+	sampleLogList := sampleLogList2(t)
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			groups, err := policy.LogsByGroup2(test.cert, sampleLogList)
+			if diff := pretty.Compare(test.want, groups); diff != "" {
+				t.Errorf("LogsByGroup: (-want +got)\n%s", diff)
+			}
+			if err != nil {
```

Comment:
nit: is it better to have this block above the diff block? (would the fact that an error was logged help to explain the diff?)

Reply:
Swapped